### PR TITLE
Stokesian Dynamics

### PR DESCRIPTION
**Pull request by jngrad**
Add a Stokesian Dynamics integrator with thermalization.

### CODE REVIEW

**Comment by jngrad** [comment]
@hmenke @biermanncarl @mkuron  This integrates all contributions from #3445 and #3241, without the SD library code.

@hmenke once you've extracted the SD library from #3445 and published it in a stand-alone repository, simply post the link here and I will update the path in the FetchContent command.

File: doc/sphinx/running.rst [hunk 1]
```diff
@@ -273,3 +273,70 @@ This convenience function only exists for historical reasons and remains availab
...
+.. _Stokesian Dynamics:
+
+Stokesian Dynamics
+------------------
+
+:meth:`espressomd.integrate.IntegratorHandle.set_stokesian_dynamics`
+
+The Stokesian Dynamics method allows to study the behaviour of spherical
+particles in a viscous fluid. It is targeted at systems with very low Reynolds
+numbers. In such systems, particles stop moving immediately as soon as any
```
Comment (RudolfWeeber) [inline doc/sphinx/running.rst]: Not a property of the system but an assumption/approximation?

File: doc/sphinx/running.rst [hunk 2]
```diff
@@ -273,3 +273,70 @@ This convenience function only exists for historical reasons and remains availab
...
+force on them is removed. In other words, motion has no memory of the past.
+
+The integration scheme is relatively simple. Only the particle's positions,
```
Comment (RudolfWeeber) [inline doc/sphinx/running.rst]: positions'?

File: doc/sphinx/running.rst [hunk 3]
```diff
@@ -273,3 +273,70 @@ This convenience function only exists for historical reasons and remains availab
...
+will fail. Therefore, the particle centers must be kept apart from each
+other by a strongly repulsive potential, for example the WCA potential
+that is set to the appropriate particle radius (for more information about
+the available interaction types see :ref:`Non-bonded interactions`).
+
+The current implementation of SD only includes the far field approximation.
+The near field (so-called lubrication) correction is planned. For now,
+Stokesian Dynamics provides a good approximation of the hydrodynamics
+in dilute systems where the average distance between particles is several
+sphere diameters.
```
Comment (RudolfWeeber) [inline doc/sphinx/running.rst:347]: Should it be stated that SD is provided via an external library.

File: src/core/unit_tests/reaction_ensemble_utils_test.cpp [hunk 4]
```diff
@@ -39,8 +39,8 @@ BOOST_AUTO_TEST_CASE(average_list_of_allowed_entries_test) {
   using namespace ReactionEnsemble;
   constexpr double tol = 100 * std::numeric_limits<double>::epsilon();
 
-  BOOST_CHECK_CLOSE(
```
Comment (RudolfWeeber) [inline src/core/unit_tests/reaction_ensemble_utils_test.cpp:42]: Why is RE stuff in this diff?

**Review by RudolfWeeber — CHANGES_REQUESTED** [review]

File: doc/sphinx/running.rst (hunk 1 above)
Comment (hmenke) [reply doc/sphinx/running.rst]: In the limit of zero Reynolds number the statement is exact.  Maybe a better formulation would be
```suggestion
numbers. In such systems, particles come to a rest almost immediately as soon as any
```

File: doc/sphinx/running.rst (hunk 2 above)
Comment (hmenke) [reply doc/sphinx/running.rst]: Did you mean `particles'`?
```suggestion
The integration scheme is relatively simple. Only the particles' positions,
```

File: src/core/unit_tests/reaction_ensemble_utils_test.cpp (hunk 4 above)
Comment (jngrad) [reply src/core/unit_tests/reaction_ensemble_utils_test.cpp:42]: there is an issue in the RE unit test that was not caught by CI on the dev branch (it was detected by the manual `intel:19` job)

File: doc/sphinx/running.rst (hunk 3 above)
Comment (jngrad) [reply doc/sphinx/running.rst:347]: I've just added that external dependency info in both the SD integrator and SD thermostat documentation.

**Review by RudolfWeeber — DISMISSED** [review]

**Comment by jngrad** [comment]
friendly ping

**Review by RudolfWeeber — APPROVED** [review]

**Pull request merged**